dev-requirements.txt

```diff
@@ -0,0 +1,3 @@
+pytest
+parameterized
```

Comment:
mostly because I couldn't figure out how to combine `pytest` parametrization and class based unit tests 🤦‍♂  and I remembered using `parameterized` package. I can give it another shot

Reply:
Okay. If remembering how to make pytest do it right takes effort, though, I don't think adding parameterized is a problem...

Reply:
https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses, I feel like I read it already. there's this hack: https://github.com/testing-cabal/unittest-ext/blob/master/params.py